system/src/main.cpp

```diff
@@ -735,6 +735,11 @@ void app_setup_and_loop(void)
     }
 #endif // HAL_PLATFORM_ASSETS
 
+    // Reset all persistent settings to factory defaults if necessary
+    // This should be called before system_part2_post_init() so that the
+    // configurations set in STARTUP() is not reset by factory reset
+    resetSettingsToFactoryDefaultsIfNeeded();
```

Comment:
Just wondering whether this makes sense or not. If not, I'll revert the change.

Reply:
Yes, it is the current behavior herein. If user is going to do a factory reset, then it doesn't make perfect sense that user app wants to restore something old from the DCT, but it would make more sense that user may want to set some new configurations from a brand new DCT, i.e after factory reset？